### PR TITLE
Defensive Stance statistic for damage prevented (DRPS/Total)

### PR DESCRIPTION
**Pull request by Cyniikal**
Defensive stance statistic module added to show DRPS prevented and total damage prevented.

### CODE REVIEW

**Comment by Cyniikal** [comment]
Okay, I'll rewrite it.

**Comment by Cyniikal** [comment]
Leaving the shield tooltip icon below the statistic in so I can also add a damage LOST due to Defensive Stance floating to the right of it.

File: src/Parser/Warrior/Arms/Modules/Talents/DefensiveStance.js [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+      return;
+    }
+    this.addEventListener('damage', this.handleDamageTaken, {
+      toPlayer: true,
+    });
+  }
+
+  handleDamageTaken(event) {
+    if(this.selectedCombatant.hasBuff(SPELLS.DEFENSIVE_STANCE_TALENT.id)) {
+      this.totalDamageMitigated += (event.mitigated || 0);
```
Comment (joshinat0r) [inline src/Parser/Warrior/Arms/Modules/Talents/DefensiveStance.js]: Are you sure about adding all the mitigated damage during def stance? I would assume that this stat shows me how much def stance did for me, not how much damage I mitigated during it. The mitigated-prop from WCL includes values from all sources (like externals, avoidance, armor, etc).

You could use something like this (just ripped that from an old version of boneshield, but you should get the idea on how we calculate the DR for just those spells):
```
const preMitigatedBoneShield = (event.amount + event.absorbed) / (1 - BONE_SHIELD_DR);
this.boneShieldMitigated += preMitigatedBoneShield * BONE_SHIELD_DR;
```

**Comment by joshinat0r** [comment]
Thanks 👍

**Pull request merged**